### PR TITLE
Fix/706 lesson 11 test bug

### PR DESCRIPTION
**Pull request by phealy3330**
**Please check if the PR fulfills these requirements:**

- [x ] The commit message follows our guidelines. (Has the prefix fix and linked to #706 on each message)

- For bug fixes and features:
    - [x ] You tested the changes.
 The following test cases were run through the regexp, though not exhaustive is fairly comprehensive:

> rotate(2) #yes
> rotate( 2 ) #yes
> rotate(   2    ) #yes
> rotate(2) #yes
> rotate(2 delta ***   ) #yes
> rotate(2 delta    ) #yes
> rotate(2    ) #yes
> rotate(2.0 delta ***   ) #yes
> rotate(2.0) #yes
> rotate(2.00) #yes
> rotate(2.000000) #yes
> rotate(2.0000000) #yes
> rotate(  2.0 ) #yes
> rotate(2.00  ) #yes
> rotate(2.000000) #yes
> rotate(2.0000000) #yes
> rotate( 2) #yes
> rotate( 2 delta ***   ) #yes
> rotate( 2 delta    ) #yes
> rotate( 2    ) #yes
> rotate( 2.0 delta ***   ) #yes
> rotate( 2.0) #yes
> rotate( 2.00) #yes
> rotate( 2.000000) #yes
> rotate(** delta   2.0000000) #yes
> rotate(     2.0 ) #yes
> rotate(    2.00  ) #yes
> rotate(delta  2.000000) #yes
> rotate( delta 2.0000000) #yes

> rotate() 2 #no
> rotate(2.) #no
> rotate() 2 ) #no
> rotate()2 ) #no
> rotate()2) #no
> rotate(2.01) #no
> rotate(2.00001) #no
> rotate(2.001) #no
> rotate(2.0001) #no
> rotate(200) #no
> rotate(.2) #no
> rotate(.20) #no
> rotate(.200)#no
> rotate(2.1)#no
> rotate(2.5)#no
> rotate(20)#no
> rotate(25)#no
> rotate(2 delta *** 05 50  ) #no
> rotate(2 100    ) #no
> rotate( 1..2    ) #no
> rotate( 12    ) #no
> 

Related issue (if applicable): #706 

**What kind of change does this PR introduce?**
Adds two Regular expressions to the Lesson 11 Practice 1 tests so that more edge cases are caught

**Does this PR introduce a breaking change?**
Not to my knowledge

**Other information**
This does actually exclude the technically correct solution: 
`rotate(2*delta*1)` and any combination that satisfies the mathematical identity property `rotate(  1 * 2 * delta)`, `rotate(0+0+1*delta*2)` etc. 

Because the regular expression that matches for 2 or 2.0 will NOT match if any other numbers are detected, other than trailing zeros in the 2.0 case.

 I think this is an acceptable tradeoff, since we're not dealing with symbolic math software and as anybody doing fancy math equations probably knows what they are doing to push the limits and break the check code. Not many "average" users are going to try something like that.

### CODE REVIEW

File: course/lesson-11-time-delta/rotating-using-delta/TestsRotatingDelta.gd [hunk 1]
```diff
@@ -22,20 +22,89 @@ func _prepare() -> void:
 
 func test_rotating_character_is_time_dependent() -> String:
 	if not _has_proper_body:
+		#reverse find for any delta at all
 		var has_delta: bool = _body.rfind("delta") > 0
```
Comment (NathanLovato) [inline course/lesson-11-time-delta/rotating-using-delta/TestsRotatingDelta.gd:26]: You can use type inference in cases like these:

```gd
var has_delta := _body.rfind("delta") > 0
```
We recommend that because the bool type is implied by the expression, and it'll be registered by Godot which will be able to document this variable as being a bool

File: course/lesson-11-time-delta/rotating-using-delta/TestsRotatingDelta.gd [hunk 2]
```diff
@@ -22,20 +22,89 @@ func _prepare() -> void:
...
-		var is_in_parentheses: bool = _body.rfind(")")
+		#set parentheses found false by default
+		var is_in_parentheses: bool = false
 
+		#Regular expression to check if delta is inside parenthesis
+		#I know, regexp is ancient evil magic, as the quote goes:
```
Comment (NathanLovato) [inline course/lesson-11-time-delta/rotating-using-delta/TestsRotatingDelta.gd:31]: Regexes are perfectly fine! Next time could you remove the comments? We can read regexes and the variable name already documents what it does

**Comment by NathanLovato** [comment]
Thank you very much for the pull request, these contributions are much appreciated!

I refactored the code a little bit. It makes a lot of sense to use a regular expression here. The expression's design makes it a little more complicated than it needs to be, but it's fine. It works, so no need to change it at this point.

You can find my changes in the last commit, don't hesitate to add comments if you have any questions

**Comment by NathanLovato** [comment]
Last note regarding commit messages, we use `prefix: user-friendly message here` The prefix is typically `fix:`, `content:`, `improvement:`, `feat:`

For example here: `fix: allow writing decimal number in lesson 11 practice`
